### ISSUE TITLE
Better signal names for buffering

### ISSUE THREAD

**Issue by raghup17** [raghup17/plasticine#26]
The 'swap' signal names confuse me every time I revisit these signal names after a while. This is minor, but the confusion mostly arises because the 'swap' signals are really 'next' signals; they don't really swap anything. Can we rename this to something else? A few suggestions:

1. Use 'next': swapWrite -> writeNext, swapRead -> readNext.
2. FIFO-centric view: swapWrite -> sramEnq, swapRead: sramDeq. Similar names for scalar FIFO.
3. Use swapRead -> readDone, swapWrite -> writeDone, since we already use 'done' signal names.

Open to discussing other names.

**Comment by yaqiz01** [comment]
I totally agree swap is not the best name. I actually internally name them incReadPtr and incWritePtr but they don't sound nice. Since sramEnq and readDone essentially are the same thing, which is the whatever that moves the pointer for multi buffering either at fine grain (FIFO) or course grain level (SRAM), I think I like `readNext` and `writeNext` are the better among these names.
